git/git.go

```diff
@@ -245,7 +245,8 @@ func ResolveRef(ref string) (*Ref, error) {
 
 	if len(lines) == 1 {
 		// ref is a sha1 and has no symbolic-full-name
-		fullref.Name = lines[0] // fullref.Sha
+		fullref.Name = lines[0]
+		fullref.Sha = lines[0]
```

Comment:
That looks sensible. I wonder, if it would make sense to check here if `lines[0]` is not empty and/or contains a 40 char string? However, in a quick check I wasn't able to make `rev-parse` return an empty result with positive error code (I guess this is not possible).

Reply:
The only thing that could cause confusion in the future is that `fullref.Name` is a hash. But that has been there before. So we are not worse than before I guess 😄